### PR TITLE
Possible changes for #326

### PR DESCRIPTION
**Pull request by MarkSummerville**
Feel free to pick parts/discard  - just wanted to put forward some solutions to issues in #326 rather than throw in problems!

### CODE REVIEW

**Comment by JoeBiellik** [comment]
Thanks, I'll give this a proper review - there are a number of little issues with the metadata generation that I'm aware of but skipped because I was focused on getting it out there. Now seems like a good time to get these features finished and polished though.

One thing I found is that Kodi itself generates NFOs that are slightly different to how their docs define it. For posterity I've attached NFO output from the latest Kodi, v2 and v3 (not inc this PR):

[kodi - tvshow.nfo](https://github.com/TV-Rename/tvrename/files/1713434/kodi.-.tvshow.txt)
[v2 - tvshow.nfo](https://github.com/TV-Rename/tvrename/files/1713435/v2.-.tvshow.txt)
[v3 - tvshow.nfo](https://github.com/TV-Rename/tvrename/files/1713436/v3.-.tvshow.txt)

[kodi - Game of Thrones - S01E01 - Winter Is Coming.nfo](https://github.com/TV-Rename/tvrename/files/1713437/kodi.-.Game.of.Thrones.-.S01E01.-.Winter.Is.Coming.txt)
[v2 - Game of Thrones - S01E01 - Winter Is Coming.nfo](https://github.com/TV-Rename/tvrename/files/1713438/v2.-.Game.of.Thrones.-.S01E01.-.Winter.Is.Coming.txt)
[v3 - Game of Thrones - S01E01 - Winter Is Coming.nfo](https://github.com/TV-Rename/tvrename/files/1713439/v3.-.Game.of.Thrones.-.S01E01.-.Winter.Is.Coming.txt)

**Comment by JoeBiellik** [comment]
Few general notes:
I've not actually seen ``<status>`` set by Kodi
We need to store more details on actors
At show level should we fill ``<episode>`` and ``<season>``? Might be useful for 3rd party scrapers
Episode NFOs should have ``showtitle``
Do we need ``premiered`` and ``year`` in ep NFOs?

File: TVRename.Core/Actions/KodiEpisodeAction.cs [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+                {
+                    Indent = true,
+                    Encoding = Encoding.UTF8,
+                    NewLineChars = "\r\n",
+
+                    //Multipart NFO files are not actually valid XML as they have multiple episodeDetails elements
+                    ConformanceLevel = ConformanceLevel.Fragment
+                });
+
+                writer.WriteStartDocument(true);
```
Comment (JoeBiellik) [inline TVRename.Core/Actions/KodiEpisodeAction.cs]: This errors out if ``ConformanceLevel.Fragment`` is used - maybe try https://stackoverflow.com/questions/19045921/net-xmlserialize-throws-writestartdocument-cannot-be-called-on-writers-created

File: TVRename.Core/Models/ProcessedEpisode.cs [hunk 2]
```diff
@@ -1,10 +1,18 @@
+using System.Collections.Generic;
 using Alphaleonis.Win32.Filesystem;
 using TVRename.Core.Models.Cache;
 
 namespace TVRename.Core.Models
 {
     public class ProcessedEpisode : Episode
     {
+        //TODO_MS Implement these methods/populate the values
```
Comment (JoeBiellik) [inline TVRename.Core/Models/ProcessedEpisode.cs]: A small thing, if you format TODOs as ``// TODO: msg - author`` Resharper indexes it nicely (Ctrl+Alt+D).

**Comment by JoeBiellik** [comment]
@MarkSummerville Are you ready to merge this?

**Comment by MarkSummerville** [comment]
Sorry - only just seen your comments - didn't get the notification for some reason. I'll fix this up later tomorrow. If it's holding you up then feel free to close this PR and grab any code that is useful

**Comment by MarkSummerville** [comment]
Hi,

Re actors then I agree: SHould be covered in this ticket: http://ideas.theideawall.com/TVRename/Forum/TopicDetails/7a02a055-d7fd-403b-898d-f8ccdeca7249

Will fix up the other stuff now

**Comment by MarkSummerville** [comment]
I have now populated:  ``<episode>``, ``<season>``, ``showtitle``, ``premiered`` and ``year`` in episode NFOs?

I think actors is a further ticket as we need to focus on being feature complete vs v2 for now.

**Pull request merged**